spec/graphql/language/parser_spec.rb

```diff
@@ -95,6 +95,18 @@
     }
   end
 
+  it "raises a parse error for invalid type modification on inline fragment spread" do
+    assert_raises(GraphQL::ParseError) {
+      GraphQL.parse("{ ... on Foo! { id } }")
```

Comment:
Should we test fragment spread as well? `{ ... Foo! }`

Reply:
We can def test it, the main bug originated from using `type` everywhere which is valid for SDL and VariableDefinitions but not for `typeCondition` which isn't being used here.